akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala

```diff
@@ -267,11 +267,8 @@ sealed trait TraversalBuilder {
   protected def internalSetAttributes(attributes: Attributes): TraversalBuilder
 
   def setAttributes(attributes: Attributes): TraversalBuilder = {
-    if (attributes ne Attributes.none) {
-      if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
-      else internalSetAttributes(attributes)
-    } else
-      this
+    if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
```

Comment:
ticket created, fill in more details if you like https://github.com/akka/akka/issues/22498